cogs/moderator.py

```diff
@@ -0,0 +1,31 @@
+import discord
+from discord.ext import commands
+
+
+class Moderator:
+    """Moderator command utilities."""
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command()
+    async def ping(self, ctx):
+        """A ping function for moderators."""
+        await ctx.send('pong')
+
+    async def __local_check(self, ctx):
+        """A local check for moderator role."""
+        if not isinstance(ctx.channel, discord.abc.GuildChannel):
+            return False
+
+        role = discord.utils.get(ctx.author.roles, name='Mod')
```

Comment:
This is basically how `commands.has_role` is implemented. However, I'm checking for a *hardcoded* `'Mod'` role here.